src/build-scripts/gh-installdeps.bash

```diff
@@ -86,14 +84,13 @@ else
         time sudo apt-get install -y g++-11
     fi
 
-    if [[ "$CXX" == "icpc" || "$CC" == "icc" || "$USE_ICC" != "" ]] ; then
+    if [[ "$CXX" == "icpc" || "$CC" == "icc" || "$USE_ICC" != "" || "$USE_ICX" != "" ]] ; then
```

Comment:
Is this missing checks for icx?
```suggestion
    if [[ "$CXX" == "icpc" || "$CC" == "icc" || "$USE_ICC" != ""  || "$CXX" == "icpx" || "$CC" == "icx" || "$USE_ICX" != ""  ]] ; then
```

Reply:
I don't use USE_ICX.

Ideally, all we should do is set CXX=icpc CC=icc, right? But for icc, I was (at least at first) having trouble building the dependencies with icc, which of course would happen if we set CXX and CC. So I switched to a method of setting USE_ICC, and in gh-installdeps.bash, that is the signal, including to switch CXX and CC at the last minute, after all dependencies are built but before we move on to building the main package.

None of the appeared necessary for icx, so I skipped the whole charade and just directly set CXX=icpx and let everything build that way.

I should go back and double check icc again. It may be that a better approach is to use icc on the dependencies, but instead make sure they are not using -Werror (or patch those projects to fix the icc-specific errors). But I was saving that for another day.